tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -328,6 +328,19 @@ class CGMSHLSLRuntime : public CGHLSLRuntime {
 };
 } // namespace
 
+static uint32_t GetIntConstAttrArg(ASTContext &astContext, const Expr *expr,
+                                   uint32_t defaultVal = 0) {
+  if (expr) {
+    llvm::APSInt apsInt;
+    APValue apValue;
+    if (expr->isIntegerConstantExpr(apsInt, astContext))
+      return (uint32_t)apsInt.getSExtValue();
+    if (expr->isVulkanSpecConstantExpr(astContext, &apValue) && apValue.isInt())
+      return (uint32_t)apValue.getInt().getSExtValue();
+  }
```

Comment:
```suggestion
    llvm_unreachable("Expression must be a constant expression or spec constant")
  }
```

We should be erroring in Sema if this isn't a constant expression or spec constant.